wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+
+                # *** depending on the overvoltage the peak dimension with respect to the noise makes changes needed on the second value. A more efficient way would be to create a bisection and stop it when it detects only the 2 biggest peaks.
+                #The number of found peaks and their positions are written into the members fNpeaks and fPositionX
+                #Search 	( 	const TH1 *  	hin,
+	        #Double_t  	sigma = 2,
+		#Option_t *  	option = "",
+		#Double_t  	threshold = 0.05 #Peaks with  amplitude < th*Amax are discarded. If too small -> detects peaks after the 511 peak. The code right now detects the last peak (that doesn't have a huge width) as the 511 peak. Other option is to iterate through the y positions and keep always the latest peak with an increase over the previous ones? first derivative with respect to max/min closest to zero 	
+                h.GetListOfFunctions().Clear()
+                spec = ROOT.TSpectrum()
+                if(args.BV==42):
+                    n_pks = spec.Search(h, 1, "", 0.01)
+                    #n_pks = spec.Search(h, 1, "", 0.999)
+                elif(args.BV>42):
+                    n_pks = spec.Search(h, 1, "", 0.005)
+                else:
+                    # print("Wrong BV<42")
+                    sys.exit("Wrong BV<42")
+
+
+                x_pos = spec.GetPositionX()
+                y_pos = spec.GetPositionY()
+
+                x_pos = np.array([x_pos[i] for i in range(n_pks)])
+                y_pos = np.array([y_pos[i] for i in range(n_pks)])
+                print("Peak charge full list:")
+                print(x_pos)
+                ind = np.argsort(x_pos)
+                x_pos = x_pos[ind]
+                y_pos = y_pos[ind]
+                                      
+                n = len(x_pos)
+                
+                # peak = x_pos[-1]
+               
+                f1 = None
+                win = 0.3 * h.GetStdDev()
+                for i in range(n-1, -1, -1):
+                    peak = x_pos[i]
+                    if peak < x_pos[0] + 0.01*h.GetStdDev():
+                        continue
+                    print(f'Fitting this peak! {peak}')
+                    f1 = ROOT.TF1("f1","gaus", peak-win, peak+win)
+                    r = h.Fit(f1, 'ILMSR+')
+                    r = r.Get()
+                    if not r.IsValid():
+                        continue
+                    if f1.GetParameter(0) > h.GetEntries() / 200:
+                        break
+                if f1 == None:
+                    peak = x_pos[-1]
+                    f1 = ROOT.TF1("f1","gaus", peak-win, peak+win)
+                    h.Fit(f1, 'ILMSR+')
+
+
+                #*** Store values in txt/csv file
+                print("511 peak = %.4f +/- %.4f" % (f1.GetParameter(1), f1.GetParError(1)))
+                save_as_csv('sodium', f1.GetParameter(1), f1.GetParError(1), h.GetName())
```

Comment:
I would add a command line argument specifying the csv file to save to and not save if nothing is provided. Eventually we want to save all these to the database and not to a csv file so that will make things easier in the future.